pyproject.toml

```diff
@@ -266,6 +266,7 @@ dl = [
   'FrEIA; python_version < "3.12"',
   'tensorflow<=2.14,>=2; python_version < "3.12"',
   'torch; python_version < "3.12"',
+  'transformers[torch]; python_version < "3.12"',
```

Comment:
@fkiraly given your comment on the dask PR, I am now very confused how this is not failing. This PR shows all tests (so installations) as passing. 

(For context to @benHeid, it was suggested in #6282 that extra marker is required. )

Reply:
I am as confused as you are. I thought this would not work, so why does it?

I explicitly checked that the `dl` dep set is actually used, and it is, in `test-full`.